src/Lecture3.hs

```diff
@@ -178,7 +207,12 @@ together only different elements.
 Product {getProduct = 6}
 
 -}
-appendDiff3 = error "TODO"
+appendDiff3 :: (Eq a, Monoid a) => a -> a -> a -> a
```

Comment:
You can notice here that you don't need to have a stricter constraint `Monoid` here and you can relax it to `Semigroup`. You don't use `mempty` so there's no need to have `Monoid` 🙂 

Your current implementation doesn't work with a type like `List1` because it doesn't have a `Monoid` instance but there's no reason why it shouldn't. So if you change the constraint to `Semigroup`, `appendDiff3` will now work with `List1`
```suggestion
appendDiff3 :: (Eq a, Semigroup a) => a -> a -> a -> a
```